From: "Gabriel Ivăncescu" <gabrielopcode@gmail.com>
Subject: [PATCH resend 1/2] comctl32: Set the selection if it's currently invalid in HandleTimer.
Message-Id: <5097375714dd96c10a4b2020ca5cda1f06d3fcf7.1606394181.git.gabrielopcode@gmail.com>
Date: Thu, 26 Nov 2020 14:37:20 +0200

Don't skip MoveCaret if it would actually change the selection when it is
invalid. This can happen, for example, in a combo box if the dropdown is shown
by a mouse click + release followed by the mouse being moved into the dropped
listbox, when the listbox has nothing selected. In this case, the item with
the index zero would not be selected the first time the mouse moves over it,
since the focus_item would be zero, despite the fact the item is not selected.

Signed-off-by: Gabriel Ivăncescu <gabrielopcode@gmail.com>
---
 dlls/comctl32/listbox.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/comctl32/listbox.c b/dlls/comctl32/listbox.c
index b905bee..299c414 100644
--- a/dlls/comctl32/listbox.c
+++ b/dlls/comctl32/listbox.c
@@ -2333,7 +2333,7 @@ static LRESULT LISTBOX_HandleTimer( LB_DESCR *descr, INT index, TIMER_DIRECTION
     case LB_TIMER_NONE:
         break;
     }
-    if (index == descr->focus_item) return FALSE;
+    if (index == descr->focus_item && descr->selected_item != -1) return FALSE;
     LISTBOX_MoveCaret( descr, index, FALSE );
     return TRUE;
 }

-- 
2.21.0

